### PR TITLE
James Burke ~ HW3 Regular Submission

### PR DESCRIPTION
**Pull request by jamesburke4**
Regular submission

### CODE REVIEW

File: HW-3.Rmd [hunk 1]
```diff
@@ -79,9 +81,52 @@ prediction of sex for all 56,946 test users, then we use the `is_female` outcome
...
+    sex = as.factor(sex),
+    # for visual purposes in charts
+    sex = fct_recode(sex, 
+                    "Male" = "m",
+                    "Female" = "f"
+                )
+  )
+
+# Creating training and test
+training <- q1_profiles[sample(1:nrow(q1_profiles), 2997, replace=FALSE),] %>%
```
Comment (rudeboybert) [inline HW-3.Rmd]: consider using `dplyr::sample_n()` here

File: HW-3.Rmd [hunk 2]
```diff
@@ -92,22 +137,48 @@ Train the logistic regression model to predict sex. i.e. fit a logistic
...
+predict_sex_model <- glm(is_female ~ income + job + orientation + body_type, 
+                         data = training, family = "binomial")
+
+# How well does the model perform on training?
+training <- training %>% 
+  mutate(p_hat1 = fitted(predict_sex_model))
 
+training <- training %>% 
+  mutate(predict_female = if_else(p_hat1 > 0.5, 1, 0)) %>% 
+  mutate(predict_correct = ifelse(predict_female == is_female, 1, 0))
```
Comment (rudeboybert) [inline HW-3.Rmd]: here 0.5 is known as a **decision threshold**

Comment (rudeboybert) [inline HW-3.Rmd]: think about

* 0.9 for advertising women-specific products
* 0.1 for absolutely including all women, even if it means including some men.

File: HW-3.Rmd [hunk 3]
```diff
@@ -24,10 +26,10 @@ library(Quandl)
...
 
 * Who you collaborated with:
-* Roughly how much time you spent on this HW so far:
-* The URL of the RPubs published URL [here](https://www.nhl.com/).
-* What gave you the most trouble:
-* Any comments you have:
+* Roughly how much time you spent on this HW so far: 7 hours
+* The URL of the RPubs published URL [here](http://rpubs.com/jamesburke4/HW-3).
+* What gave you the most trouble: 
+* Any comments you have: Problem occurs when training doesn't include a factor but test does. For instance, I received an error when using test data with model because there was no one in the training dataset that reported income of 500,000.
```
Comment (rudeboybert) [inline HW-3.Rmd]: yeah, this is a problem when categorical variables are too specific. you generally want to set them in a way so that they bucket a lot of observations OR set up your code so that a "new" unseen category gets assigned to an "other" category to make your model more robust.

File: HW-3.Rmd [hunk 4]
```diff
@@ -92,22 +137,48 @@ Train the logistic regression model to predict sex. i.e. fit a logistic
...
+# How well does the model perform on training?
+training <- training %>% 
+  mutate(p_hat1 = fitted(predict_sex_model))
 
+training <- training %>% 
+  mutate(predict_female = if_else(p_hat1 > 0.5, 1, 0)) %>% 
+  mutate(predict_correct = ifelse(predict_female == is_female, 1, 0))
+
+# The proportion of predictions that are correct
+round(mean(training$predict_correct), 5)
```
Comment (rudeboybert) [inline HW-3.Rmd]: yes, I expected a larger difference between training and test performance. In your case, your model was not too overfit. Also keep in mind, there is a bit of variation due to what observations get assigned to your test/training data depending on what seed value you use in set.seed.
